core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/tuple/Tuple.java

```diff
@@ -580,4 +584,56 @@ public Builder removeIfExists(String... attributes) {
         }
 
     }
+
+    public static class BuilderV2 {
```

Comment:
I very much like the idea of having a V2 version. Hope it will solve our memory issues in some ways. I am thinking about this name `fill`. I haven't seen it before. Eg: If an array already has an element at position x, we replace it by saying arr.add(x,value). 

I still support having the name `add`. Our exceptions and comments will take care of anything wrong. Any thoughts? What names do other builders use?

Reply:
I was thinking more from the perspective of "we already have the designated slots and now just filling them in". Also, there are similar APIs in Java [`Arrays#fill`](https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#fill(java.lang.Object[],%20java.lang.Object)) and JavaScript [`Array#fill`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill).

But on second thoughts, those `fill` APIs just add a range of values or a default value. It does give rise to a different semantic. I'll change to `add` itself, and let the comments and more importantly the exceptions do the major talking as suggested.